### PR TITLE
Feature/load course swen900132021 ag 93

### PR DESCRIPTION
**Pull request by Ev-anz**

### CODE REVIEW

**Review by YIHDONG — APPROVED** [review]
looks good to me, I gonna approve and merge it.

File: src/util/algorithms.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[
+    {
```
[the remark this answers is not in the document]
Comment (Ev-anz) [reply src/util/algorithms.json:2]: Will remove it in a following commit

File: src/courses/course1/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const page1 = require('./page1');
+const page2 = require('./page2');
+
+module.exports = {
+  name: 'dijkstra',
+  id: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
```
[the remark this answers is not in the document]
Comment (Ev-anz) [reply src/courses/course1/index.js:6]: I've tried to use UUID for each of the courses, the problem is I cannot get past the unit tests if I use a generated id, I guess I need some more time to resolve this

File: src/util/loadCourses.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+const courseList = require('../courses');
+
+class CourseLoader {
+  constructor() {
+    this.courselist = courseList;
+  }
+
+  /* get a course by name */
+
+  // Currently I'm not so sure if we need this function, will just leave it here
```
[the remark this answers is not in the document]
Comment (Ev-anz) [reply src/util/loadCourses.js:10]: Will remove it in a following commit

File: src/util/loadCourses.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      if (course.id === id) {
+        res = true;
+      }
+    });
+    return res;
+  }
+
+  /* get a course by id */
+  getCourseById(id) {
+    let res = {};
```
[the remark this answers is not in the document]
Comment (Ev-anz) [reply src/util/loadCourses.js:42]: The controller uses the hasCourse() function to check the existence of a course, getCourseById only gets called after the existence is proved to be true

**Pull request merged**